src/libraries/Native/Unix/System.Globalization.Native/pal_collation.c

```diff
@@ -38,8 +38,16 @@ c_static_assert_msg(USEARCH_DONE == -1, "managed side requires -1 for not found"
 // change ICU's default behavior here isn't really justified unless someone has a strong reason
 // for !StringSort to behave differently.
 
+#define USED_STRING_SEARCH ((UStringSearch*) (-1))
+
 typedef struct { int32_t key; UCollator* UCollator; } TCollatorMap;
 
+typedef struct SEARCHNODE
+{
+    UStringSearch* searchIterator;
+    struct SEARCHNODE* next;
+} SearchIteratorNode;
```

Comment:
a nit: it can be simplified to:

```c
typedef struct SearchIteratorNode
{
    int* searchIterator;
    struct SearchIteratorNode* next;
};
```

Reply:
@am11, your suggestion causing build breaks like the following:

```
 /root/oss/runtime/src/libraries/Native/Unix/System.Globalization.Native/pal_collation.c:45:1: error: typedef requires a name [-Werror,-Wmissing-declarations]
  typedef struct SearchIteratorNode
  ^~~~~~~
  /root/oss/runtime/src/libraries/Native/Unix/System.Globalization.Native/pal_collation.c:59:5: error: must use 'struct' tag to refer to type 'SearchIteratorNode'
      SearchIteratorNode searchIteratorList[CompareOptionsMask + 1];
      ^
      struct
  /root/oss/runtime/src/libraries/Native/Unix/System.Globalization.Native/pal_collation.c:396:17: error: must use 'struct' tag to refer to type 'SearchIteratorNode'
                  SearchIteratorNode* pNext = pSortHandle->searchIteratorList[i].next;
                  ^
                  struct
  /root/oss/runtime/src/libraries/Native/Unix/System.Globalization.Native/pal_collation.c:405:21: error: must use 'struct' tag to refer to type 'SearchIteratorNode'
                      SearchIteratorNode* pCurrent = pNext;
                      ^
                      struct
  /root/oss/runtime/src/libraries/Native/Unix/System.Globalization.Native/pal_collation.c:451:5: error: must use 'struct' tag to refer to type 'SearchIteratorNode'
      SearchIteratorNode* node = (SearchIteratorNode*) malloc(sizeof(SearchIteratorNode));
      ^
      struct
  /root/oss/runtime/src/libraries/Native/Unix/System.Globalization.Native/pal_collation.c:451:52: error: expected expression
      SearchIteratorNode* node = (SearchIteratorNode*) malloc(sizeof(SearchIteratorNode));
                                                     ^
  /root/oss/runtime/src/libraries/Native/Unix/System.Globalization.Native/pal_collation.c:451:33: error: use of undeclared identifier 'SearchIteratorNode'
      SearchIteratorNode* node = (SearchIteratorNode*) malloc(sizeof(SearchIteratorNode));
                                  ^
  /root/oss/runtime/src/libraries/Native/Unix/System.Globalization.Native/pal_collation.c:460:5: error: must use 'struct' tag to refer to type 'SearchIteratorNode'
      SearchIteratorNode* pCurrent = &pSortHandle->searchIteratorList[options];
      ^
      struct
  /root/oss/runtime/src/libraries/Native/Unix/System.Globalization.Native/pal_collation.c:463:5: error: must use 'struct' tag to refer to type 'SearchIteratorNode'
      SearchIteratorNode* pNull = NULL;
      ^
      struct
  /root/oss/runtime/src/libraries/Native/Unix/System.Globalization.Native/pal_collation.c:483:5: error: must use 'struct' tag to refer to type 'SearchIteratorNode'
      SearchIteratorNode* pCurrent = &pSortHandle->searchIteratorList[options];
      ^
      struct
  /root/oss/runtime/src/libraries/Native/Unix/System.Globalization.Native/pal_collation.c:536:5: error: must use 'struct' tag to refer to type 'SearchIteratorNode'
      SearchIteratorNode* pCurrent = &pSortHandle->searchIteratorList[options];
      ^
      struct
```

Reply:
Ops, fixed the typo in alias.